integration-tests/scripts/create-github-repo.sh

```diff
@@ -0,0 +1,155 @@
+if [ "$private" != "true" ] && [ "$private" != "false" ]; then
+  echo "🔴 error: private parameter must be 'true' or 'false', got: $private"
+  exit 1
+fi
+
+# Check if repo_name contains organization (has a slash)
+if [[ "$repo_name" == *"/"* ]]; then
+  # Organization repository format: org/repo
+  ORG_NAME=$(dirname "$repo_name")
+  ACTUAL_REPO_NAME=$(basename "$repo_name")
+  full_repo_name="$repo_name"
+  CREATE_URL="https://api.github.com/orgs/${ORG_NAME}/repos"
+  
+  if [ -n "${DEBUG}" ]; then
+    echo "🐛 Organization repository detected"
+    echo "🐛 Organization: ${ORG_NAME}"
+    echo "🐛 Repository name: ${ACTUAL_REPO_NAME}"
+    echo "🐛 Create URL: ${CREATE_URL}"
+  fi
+else
+  # User repository format: just repo name
+  # Get current user to construct full repo name
+  echo "Getting GitHub user information..."
+  USER_RESPONSE=$(curl -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/user 2> /dev/null)
+  if [ -n "${DEBUG}" ]; then
+    echo "🐛 User API response: ${USER_RESPONSE}"
+  fi
+  GITHUB_USER=$(echo "${USER_RESPONSE}" | jq -r '.login // ""')
+  if [ -z "${GITHUB_USER}" ]; then
+    echo "🔴 error: could not get GitHub user information"
+    if [ -n "${DEBUG}" ]; then
+      echo "🐛 Full user response: ${USER_RESPONSE}"
+    fi
+    echo "   Check that GITHUB_TOKEN is valid and has appropriate permissions"
+    exit 1
+  fi
+  if [ -n "${DEBUG}" ]; then
+    echo "🐛 GitHub user: ${GITHUB_USER}"
+  fi
+  
+  ORG_NAME="$GITHUB_USER"
+  ACTUAL_REPO_NAME="$repo_name"
+  full_repo_name="${GITHUB_USER}/${repo_name}"
+  CREATE_URL="https://api.github.com/user/repos"
+  
+  if [ -n "${DEBUG}" ]; then
+    echo "🐛 User repository detected"
+    echo "🐛 User: ${GITHUB_USER}"
+    echo "🐛 Repository name: ${ACTUAL_REPO_NAME}"
+    echo "🐛 Create URL: ${CREATE_URL}"
+  fi
+fi
+
+# Check if repository already exists
+echo "Checking if repository ${full_repo_name} already exists..."
+EXISTING_REPO_RESPONSE=$(curl -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/${full_repo_name} 2> /dev/null)
+if [ -n "${DEBUG}" ]; then
+  echo "🐛 Existing repo check response: ${EXISTING_REPO_RESPONSE}"
+fi
+EXISTING_REPO=$(echo "${EXISTING_REPO_RESPONSE}" | jq -r '.full_name // ""')
+if [ -n "${EXISTING_REPO}" ]; then
+  echo "⚠️  Repository ${full_repo_name} already exists"
+  echo "   URL: https://github.com/${full_repo_name}"
+  exit 0
+fi
+
+# Create the repository
+echo "Creating repository ${full_repo_name}..."
+echo "  - Description: ${description}"
+echo "  - Private: ${private}"
+
+if [ -n "${DEBUG}" ]; then
+  echo "🐛 Creating with payload: {\"name\":\"${ACTUAL_REPO_NAME}\",\"description\":\"${description}\",\"private\":${private}}"
+  echo "🐛 Using URL: ${CREATE_URL}"
+fi
+
+CREATE_RESPONSE=$(curl -X POST -H "Authorization: token $GITHUB_TOKEN" \
+  -d "{\"name\":\"${ACTUAL_REPO_NAME}\",\"description\":\"${description}\",\"private\":${private}}" \
+  "${CREATE_URL}" 2> /dev/null)
+
+if [ -n "${DEBUG}" ]; then
```

Comment:
The script already does `set -x` if debug is true. I think all these echoes are pretty redundant and unnecessary